### PR TITLE
feat(x/ecocredit): query `Classes` ORM

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

adds query for classes 

ref: #696 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/server/core/query_classes_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		Metadata:   nil,
+		CreditType: "C",
+	})
+	assert.NilError(t, err)
+
+	// requesting all
+	res, err := s.k.Classes(s.ctx, &v1beta1.QueryClassesRequest{})
+	assert.NilError(t, err)
+	assert.Equal(t, 2, len(res.Classes))
+	assert.Equal(t, "C01", res.Classes[0].Name)
```
Comment (ryanchristo) [inline x/ecocredit/server/core/query_classes_test.go:33]:
```suggestion
	assert.Equal(t, "C01", res.Classes[0].Name)
	assert.Equal(t, "C02", res.Classes[1].Name)
```

File: x/ecocredit/server/core/query_classes_test.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+	// request with pagination
+	res, err = s.k.Classes(s.ctx, &v1beta1.QueryClassesRequest{Pagination: &query.PageRequest{
+		Limit:      1,
+		CountTotal: true,
+		Reverse:    true,
+	}})
+	assert.NilError(t, err)
+	assert.Equal(t, 1, len(res.Classes))
+	assert.Equal(t, uint64(2), res.Pagination.Total)
```
Comment (ryanchristo) [inline x/ecocredit/server/core/query_classes_test.go:43]:
```suggestion
	assert.Equal(t, uint64(2), res.Pagination.Total)
	assert.Equal(t, "C02", res.Classes[0].Name)
```

**Review by ryanchristo — APPROVED** [review]
Looks good to me.

**Review by amaurym — APPROVED** [review]
LGTM

File: x/ecocredit/server/core/query_classes.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import (
+	"context"
+	"github.com/cosmos/cosmos-sdk/orm/model/ormlist"
+	ecocreditv1beta1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1beta1"
+	"github.com/regen-network/regen-ledger/x/ecocredit/v1beta1"
+)
+
+// Classes queries for all credit classes with pagination.
+func (k Keeper) Classes(ctx context.Context, request *v1beta1.QueryClassesRequest) (*v1beta1.QueryClassesResponse, error) {
+	pg, err := GogoPageReqToPulsarPageReq(request.Pagination)
```
Comment (anilCSE) [inline x/ecocredit/server/core/query_classes.go:12]: I think we should move `x/ecocredit/server/core/query_utils.go` to global scope `types` or `app/utils`. It smells generic.

Comment (anilCSE) [inline x/ecocredit/server/core/query_classes.go:12]: A different scope, but I think we should move `x/ecocredit/server/core/query_utils.go` to global scope `types` or `app/utils`. It smells generic.

**Review by anilCSE — APPROVED** [review]

File: x/ecocredit/server/core/query_classes.go (hunk 3 above)
Comment (technicallyty) [reply x/ecocredit/server/core/query_classes.go:12]: yeah its pretty generic, but i think we can punt it until someone needs it elsewhere

**Pull request merged**